This is very useful library. but when setMain(viewController:), it do…

oslover opened the pull request:
This is very useful library. but when setMain(viewController:), it doesn't remove the tapGesture, so after a new Main View controller is shown, user is unable to tap on the main view.
Therefore, in animateMainViewController, we need to remove Tap Gesture. or in setMain method.

oslover committed:
This is very useful library. but when setMain(viewController:), it doesn't remove the tapGesture, so after a new Main View controller is shown, user is unable to tap on the main view.
Therefore, in animateMainViewController, we need to remove Tap Gesture. or in setMain method.

oslover committed:
in line with the swift style.

oslover reviewed (commented):
in line with swift format

audrl1010 commented:
Thank you for your interest in my project .
Putting your modified source inside the `animateMainViewController ()` method does not seem right. This is because `animateMainViewController ()` is a method that only performs animation. It is recommended that you put the modified source part in the `setMain (viewController newMainViewController: UIViewController?)` Method.

audrl1010 commented:
Thank you for your interest in my project .
Putting your modified source inside the `animateMainViewController ()` method does not seem right. This is because `animateMainViewController ()` is a method that only performs animation. It is recommended that you put the modified source part in the `setMain (viewController newMainViewController:_) Method`.

audrl1010 commented:
Thank you for your interest in my project .
(currentSlideOutState! = .main) in case, It should be a tapGesture of mainContainerView. The reason is (currentSlideOutState! = .main) in this case so that when the user taps the mainViewContainer, it will return to main. Therefore, you do not have to process it in the animationViewController method.

oslover commented:
Thank you for your reply. 
but i actually experienced that it was not able to tap the tableviewcell in the mainview after the mainview was shown. i was only possible to tap the cell by second tap even after the mainview had already covered the whole iphone screen. it means the tapgesture was not removed by the moment. so i removed the tapgesture in the animateMainViewController method. 
and then, after i changed the code, i could tap the tableviewcell by first touch after the mainview appeared in the iphone screen. 
Therefore, if you don't like to add the code in animateMainViewController method, i think we need to remove the tap gesture somewhere else. 
I hope you test it by yourself. i think it's kinda serious problem.

oslover commented:
PS: it works fine if mainview is already set in the appdelegate, but not work only when the mainview is newly set by setMain(method) by selecting any item in the leftView.

audrl1010 commented:
Temporarily blocked tap on source. The reason is to prevent continuous tap.
